packages/hub/services/prisma-manager.ts

```diff
@@ -7,51 +7,58 @@ import {
 } from './prisma-extensions/push-notification-registration';
 import { ExtendedLatestEventBlock, getLatestEventBlockExtension } from './prisma-extensions/latest-event-block';
 import { ExtendedExchangeRate, getExchangeRateExtension } from './prisma-extensions/exchange-rate';
+
 export interface ExtendedPrismaClient extends PrismaClient {
   exchangeRate: ExtendedExchangeRate;
   pushNotificationRegistration: ExtendedPushNotificationRegistration;
   latestEventBlock: ExtendedLatestEventBlock;
 }
 
+let dbConfig: Record<string, any> = config.get('db');
+
+// Prisma client should be a singleton to avoid this problem:
+// https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/instantiate-prisma-client#the-number-of-prismaclient-instances-matters
+
+let singletonClient = new PrismaClient({
+  datasources: { db: { url: dbConfig.url } },
+  log: dbConfig.prismaLog,
+});
+
+// In tests, do not add extensions until the in-transaction proxy client exists
+
+if (!dbConfig.useTransactionalRollbacks) {
+  addCardstackPrismaExtensions(singletonClient);
+}
```

Comment:
Should this be in the `ready` hook for the service instead?

Reply:
sure, done in 56ee76c, thanks